### PR TITLE
[datafeeder] Add missing properties to published GeoServer layers

### PR DESCRIPTION
**Pull request by groldan**
Add AttributionInfo to published GeoServer layer, and client cache settings to the feature type

Depends on https://github.com/geoserver/geoserver/pull/5103

### CODE REVIEW

**Comment by groldan** [comment]
Waiting for geoserver's nightly build to publish the changes at `https://github.com/geoserver/geoserver/pull/5103` to maven

File: datafeeder/src/main/java/org/georchestra/datafeeder/service/publish/impl/GeorchestraOwsPublicationService.java [hunk 1]
```diff
@@ -223,9 +257,13 @@ private FeatureTypeInfo buildPublishingFeatureType(String workspace, String data
             ft.setProjectionPolicy(ProjectionPolicy.FORCE_DECLARED);
         }
 
+        // make the layer cacheable
+        MetadataMap mdmap = new MetadataMap();
+        mdmap.addEntryItem(new MetadataEntry().atKey("cacheAgeMax").value("3600"));
```
Comment (fvanderbiest) [inline datafeeder/src/main/java/org/georchestra/datafeeder/service/publish/impl/GeorchestraOwsPublicationService.java:262]: Can it be a setting ?

**Review by fvanderbiest — APPROVED** [review]
Looks good to me ! Thanks Gabriel.
So we need to patch our geoserver fork before the fixes to the rest client are released (in 2.19.x ?)

**Comment by landryb** [comment]
> Looks good to me ! Thanks Gabriel.
> So we need to patch our geoserver fork before the fixes to the rest client are released (in 2.19.x ?)

and stress somewhere in the doc that datafeeder relies on georchestra-geoserver :)

**Comment by groldan** [comment]
> So we need to patch our geoserver fork before the fixes to the rest client are released (in 2.19.x ?)
Nope, we use 20.0-SNAPSHOT already. It's safe cause we're the only ones using that community module

File: datafeeder/src/main/java/org/georchestra/datafeeder/service/publish/impl/GeorchestraOwsPublicationService.java (hunk 1 above)
Comment (groldan) [reply datafeeder/src/main/java/org/georchestra/datafeeder/service/publish/impl/GeorchestraOwsPublicationService.java]: Done: 25c8c2e

**Comment by groldan** [comment]
> `cacheSeconds.intValue() <= 0` ?

ok, b61fbaf. Merging.

**Comment by landryb** [comment]
> Nope, we use 20.0-SNAPSHOT already. It's safe cause we're the only ones using that community module

well that had lead to a build failure of master before this PR (need to recheck) so it isnt that safe :)

```
[ERROR] /data/src/georchestra/georchestra/datafeeder/src/main/java/org/georchestra/datafeeder/service/publish/impl/GeorchestraOwsPublicationService.java:[269,13] cannot find symbol
[ERROR]   symbol:   method setId(java.lang.String)
[ERROR]   location: variable info of type org.geoserver.openapi.model.catalog.MetadataLinkInfo
```

**Comment by groldan** [comment]
The question was if "we need to patch our geoserver fork before the fixes to the rest client are released (in 2.19.x ?)"
And it's safe in that regard, we don't need to patch geoserver, because we're using the client library, which doesn't require any server-side change.
It's also safe to use the -SNAPSHOT dependency instead of waiting for a stable release because, by one side, we're the only ones using that client library, and by the other, even if there were a stable geoserver release, the community module wouldn't be part of it, so we'd still be using the SNAPSHOT version.
But of course, if we needed to make API changes to the client library in order to fix this issue, we need to get it in sync (i.e. master vs master).

**Pull request merged**